### DISCUSSION TITLE
Example showing to use this in production (Windows + MSVC)

### DISCUSSION THREAD

**Discussion by SeeRich** [jeremy-rifkin/cpptrace#168] [Ideas]
Thank you for working on this amazing project! It is much better than anything else I could find.

I have been using this on a Qt/C++/Windows/MSVC GUI application.

During the development cycle, it is amazing! Anytime I have an exception, I get a pretty colored stack trace to help me find the issue.

However, I am somewhat confused about how to move this to production. Currently, I generate the debug symbols in a separate PDB file. I don't want to ship this PDB to the end users (it is huge and also contains alot of the application internal logic). How can I generate something (stacktrace, core dump, mini dump) in a production environment and then once that is reported/collected debug it using the PDB on another machine?

I hope this question makes sense.

**Comment by jeremy-rifkin** [comment]
Hi, thanks for the kind words :)

What you're trying to do is possible, though, I apologize in advance it isn't the most painless process.

What I'd do is, in production, output a `cpptrace::object_trace`. This is intended to contain the bare minimum information needed to figure out the rest of the trace details. It will have handled things like runtime address randomization and also figured out what executables/dlls the frames correspond to. On your development machine, where you have a build of the production .exe and .pdb for it, you can load that `object_trace` and have cpptrace resolve it against the exe and pdb. Some manipulation of the paths in the object trace might be needed (e.g. it might have `C:\ProgramFiles\Bar\Bar.exe` but that now needs to be `C:\Users\Foo\Projects\Bar\artifacts\Bar-1.2.2.exe`).

I'll try to add some documentation on this. I think it would also be beneficial for the library to somehow make this process less painful, though I'm not immediately sure how best to facilitate this.

Hope it helps!

**Reply by SeeRich** [reply]
This is exactly what I need!

That would be amazing if there existed some utility functions to make this easier. Let me know if there is something I can do to help (not implement but test or something).

**Reply by jeremy-rifkin** [reply]
Thanks, I appreciate the offer! If you have any ideas for utility functions that'd be helpful here I'd be interested. It's not immediately clear to me how a good interface / utility could be provided for this so I'll have to think on it more.

**Reply by RockWithBoA** [reply]
Hi @jeremy-rifkin ,

I would say cpptrace is amazing after testing it in my actual app! And I also would like to know more details about how to get stacktrace in production. 

As you said in production, we can output a cpptrace::object_trace and on development machine, we can load that object_trace and have cpptrace resolve it against the exe and pdb. Could you please provide more detailed steps to do this? Do I miss such info in the documentation?

Basically what I want to know is:
- How to output the cpptrace::object_trace?
- How to load the object_trace and resolve it against the exe and pdb? 


I'm not quite familar with the technique in getting stacktrace, so maybe asking some rookie questions :)
